TBOX-242 Check dataset is streamable snippet

macgregg1 opened the pull request:
<!---
Thanks for filing a pull request 😄 ! 

Pull requests are only accepted from Tamr employees. If there is a change you would like to see, please file a GitHub Issue or contact your Tamr Professional Services representative.

Below is a checklist of things to do before it can be merged.
-->

# ↪️ Pull Request

<!---
Provide a general summary of the pull request here
Please look for any issues that this PR resolves and tag them in the PR.
-->


## ✔️ PR Todo

- [x] Add documentation
- [x] Add examples
- [x] Ensure you are following the practices in the [contributor guide](https://docs.google.com/document/d/1c_hhpDMhO0zN793Z_HFOz42Q4g7l1nPRVQ5pXOUsMJU/edit) and [coding_standards](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards)
  * [Use consistent and descriptive name conventions](https://docs.google.com/document/d/1c_hhpDMhO0zN793Z_HFOz42Q4g7l1nPRVQ5pXOUsMJU/edit#heading=h.477s36w4rds9) 
  * [Uses google-style docstrings](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards#Google-style-docstrings)
  * [Uses versioned API/client methods only](https://docs.google.com/document/d/1c_hhpDMhO0zN793Z_HFOz42Q4g7l1nPRVQ5pXOUsMJU/edit#heading=h.yv6df99fyrq2)
  * [Avoid global variables](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards#Avoid-global-variables)
  * [Avoid giant try/catch](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards#Avoid-giant-try-/-except)
  * [Don't use mutable default variables](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards#Don%E2%80%99t-use-mutable-default-arguments)
  * [Use a `main()` function in your scripts](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards#Use-a-%E2%80%9Cmain()%E2%80%9D--function-in-your-scripts)
- [ ] Added/updated testing for this change (ensure you are testing any changes/new code!)
- [x] Update title to link to TBOX issue (i.e. start pull request title with TBOX-xx) 
- [x] Approval from first code-reviewer
- [ ] Approval from second code-reviewer (only required if a second reviewer has left comments)
- [x] Pass all checks

Commit:
Check dataset is streamable snippet

skalish commented on examples/snippets/data_io/df_connect/streamable_check.py:
```diff
@@ -0,0 +1,31 @@
+tbox.data_io.df_connect.client.export_dataset(
+    my_connect, dataset_name=dataset.name, target_table_name="target_table"
+)
+# if dataset is not streamable, this will allow it to be refreshed
```
Can you stick a newline here, just be consistent with the spacing above?
```suggestion

# if dataset is not streamable, this will allow it to be refreshed
```

skalish commented on examples/snippets/data_io/df_connect/streamable_check.py:
```diff
@@ -0,0 +1,31 @@
+
+# if dataset is not streamable, this will raise an error
+if not dataset.status().is_streamable:
+    message = f"Dataset {dataset.name} is not streamable."
+    raise RuntimeError(message)
```
There's no need to create a variable here, since you are just using it immediately:
```suggestion
    raise RuntimeError(f"Dataset {dataset.name} is not streamable.")
```

skalish commented on examples/snippets/data_io/df_connect/streamable_check.py:
```diff
@@ -0,0 +1,31 @@
+"""Snippet to add an explicit check to validate that a dataset is streamable using df_connect
+Do this by passing in a Dataset object rather than just the dataset name"""
```
I don't think this second line is necessary, also slight wording change:
```suggestion
"""Snippet to add an explicit check to validate that a dataset is streamable when exporting with df_connect"""
```

skalish reviewed (commented):
This looks pretty good to me. I've left a few comments with suggested changes.  I think this is ready after those are addressed.

Commit:
Apply suggestions from code review

Cleaning up language and error (unnecessary message). Maintaining consistent spacing.

Co-authored-by: skalish <39866163+skalish@users.noreply.github.com>

skalish commented:
You can resolve the failing lint check by running `invoke format --fix` locally.  That should get this ready to merge.

Commit:
Apply sugggestions from code review

Commit:
fixed error message

skalish commented on examples/snippets/data_io/df_connect/streamable_check.py:
```diff
@@ -0,0 +1,31 @@
+"""Snippet to add an explicit check to validate that a dataset is streamable when exporting with
+df_connect"""
```
Can this be one line?

macgregg1 replied:
yes! Just tightened language

Commit:
tightening snippet explanation

skalish reviewed (approved):
LGTM. Assuming all CI tasks complete successfully this is good to merge.
